### PR TITLE
consistent container copies

### PR DESCRIPTION
**Pull request by brauner**

### CODE REVIEW

**Comment by brauner** [comment]
Few things left:
- enable `ContainerOnly` copy for remote copies and moves
- add tests

**Comment by brauner** [comment]
New criu messes with my local tests so needed to test a little prematurely here. :)

**Comment by brauner** [comment]
@stgraber, this now allows:

- `lxc copy a b --container-only`
- `lxc copy r1:a r2:b --container-only`
- `lxc move r1:a r2:b --container-only`
- `lxc move a b` will simply rename everything as before

**Comment by stgraber** [comment]
Cool, so just missing tests then?

**Comment by brauner** [comment]
@stgraber, tests added.

**Comment by brauner** [comment]
Fixing.

**Comment by brauner** [comment]
`MigrationS{ource,ink}()` need a `containerOnly` arg.

**Comment by brauner** [comment]
TODO:
- [ ] `btrfs` backend still needs to be adapted to properly handle `btrfs` to `btrfs` backend

**Comment by brauner** [comment]
jenkins: test this please

**Comment by brauner** [comment]
@stgraber, should be good for review.

**Comment by stgraber** [comment]
yup, will do in a bit, looks like it'll take me a little while to read through it all

File: shared/api/container.go [hunk 1]
```diff
@@ -92,5 +92,6 @@ type ContainerSource struct {
 	Live bool `json:"live,omitempty" yaml:"live,omitempty"`
 
 	// For "copy" type
-	Source string `json:"source,omitempty" yaml:"source,omitempty"`
+	Source        string `json:"source,omitempty" yaml:"source,omitempty"`
+	ContainerOnly bool   `json:"container_only,omitempty" yaml:"container_only,omitempty"`
```
Comment (stgraber) [inline shared/api/container.go]: This needs to be in its own section with an API extension comment above it.

File: lxc/copy.go [hunk 2]
```diff
@@ -35,6 +36,7 @@ func (c *copyCmd) flags() {
 	gnuflag.Var(&c.profArgs, "p", i18n.G("Profile to apply to the new container"))
 	gnuflag.BoolVar(&c.ephem, "ephemeral", false, i18n.G("Ephemeral container"))
 	gnuflag.BoolVar(&c.ephem, "e", false, i18n.G("Ephemeral container"))
+	gnuflag.BoolVar(&c.containerOnly, "container-only", false, i18n.G("Copy container without snapshots"))
```
Comment (stgraber) [inline lxc/copy.go]: For consistency in sentence structure with the others, that ought to be "Copy the container without its snapshots".

File: lxc/move.go [hunk 3]
```diff
@@ -14,21 +17,23 @@ func (c *moveCmd) showByDefault() bool {
...
+lxc move <old name> <new name> [--container-only]
     Rename a local container.
 
 lxc move <container>/<old snapshot name> <container>/<new snapshot name>
     Rename a snapshot.`)
 }
 
-func (c *moveCmd) flags() {}
+func (c *moveCmd) flags() {
+	gnuflag.BoolVar(&c.containerOnly, "container-only", false, i18n.G("Move container without snapshots"))
```
Comment (stgraber) [inline lxc/move.go]: Same comment as for copy

File: lxd/migrate.go [hunk 4]
```diff
@@ -358,7 +367,8 @@ func (s *migrationSourceWs) Do(migrateOp *operation) error {
 		return fsErr
 	}
 
-	if err := s.recv(&header); err != nil {
+	err = s.recv(&header)
+	if err != nil {
```
Comment (stgraber) [inline lxd/migrate.go]: Can you please move those kind of non-functional changes to a separate commit?
This makes it hard to review and will cause a massive headache for stable-2.0 if I can't easily cherry-pick those.

**Comment by stgraber** [comment]
Before I re-review everything can you address those few comments and re-split the branch like so:
 - Non-functional code change (you are undoing a lot of one-line if statements in there)
 - Daemon side changes including documentation changes for the API extension
 - Client side changes (client.go and lxc/)
 - i18n (running both "make i18n && make update-po" to also refresh the language templates)
 - tests

That should make things easier to review and also cherry-pick the bits that make sense to 2.0.

File: shared/api/container.go (hunk 1 above)
Comment (brauner) [inline shared/api/container.go]: done

File: lxc/copy.go (hunk 2 above)
Comment (brauner) [inline lxc/copy.go]: done

File: lxc/move.go (hunk 3 above)
Comment (brauner) [inline lxc/move.go]: done

**Comment by brauner** [comment]
Sure, makes sense if we're actually backporting all of that stuff.

**Comment by brauner** [comment]
jenkins: test this please

**Comment by brauner** [comment]
@stgraber, added a lot more non-functional changes to justify the separate commit. :)

File: shared/api/container.go [hunk 5]
```diff
@@ -18,7 +18,8 @@ type ContainerPost struct {
 	Name string `json:"name" yaml:"name"`
 
 	// Used for migration
-	Migration bool `json:"migration" yaml:"migration"`
+	Migration     bool `json:"migration" yaml:"migration"`
+	ContainerOnly bool `json:"container_only" yaml:"container_only"`
```
Comment (stgraber) [inline shared/api/container.go]: Also needs to be moved under an API extension comment.

Comment (brauner) [inline shared/api/container.go]: done

**Comment by brauner** [comment]
@stgraber, should be good.

**Pull request merged**